Build failure: `python312Packages.tpm2-pytss`

## Steps To Reproduce

`nix-build -A python312Packages.tpm2-pytss`

I'm trying to bisect this now.

## Build log

<details>

<summary><code>nix-build -A python312Packages.tpm2-pytss</code></summary>


```
============================= test session starts ==============================
platform linux -- Python 3.12.8, pytest-8.3.3, pluggy-1.5.0
rootdir: /build/tpm2-pytss-2.3.0
configfile: pyproject.toml
collected 611 items

test/test_crypto.py ................................................     [  7%]
test/test_cryptography.py ...............................                [ 12%]
test/test_encoding.py .................................................. [ 21%]
..F...................                                                   [ 24%]
test/test_esapi.py .ss.................................................. [ 33%]
..s.................ss.s................................................ [ 45%]
..                                                                       [ 45%]
test/test_exception.py ......                                            [ 46%]
test/test_fapi.py ...................................................... [ 55%]
.................................................................        [ 65%]
test/test_import.py ...                                                  [ 66%]
test/test_internal.py .                                                  [ 66%]
test/test_policy.py ..............                                       [ 68%]
test/test_tcti.py s.s....s.....s..s...                                   [ 72%]
test/test_tcti_spi_helper.py ......                                      [ 73%]
test/test_tsskey.py .............                                        [ 75%]
test/test_types.py ..................................................... [ 83%]
........................................................................ [ 95%]
                                                                         [ 95%]
test/test_utils.py ...............ss.........                            [100%]

=================================== FAILURES ===================================
_____________ ToolsTest.test_tools_decode_tpml_tagged_tpm_property _____________

self = <test.test_encoding.ToolsTest testMethod=test_tools_decode_tpml_tagged_tpm_property>

    def test_tools_decode_tpml_tagged_tpm_property(self):
        if not self.has_tools:
            self.skipTest("tools not in path")
        elif not self.tpm.tcti_name_conf.startswith("swtpm"):
            self.skipTest("swtpm required")

        fixedyml = self.run_tool("getcap", "properties-fixed")
        fl = from_yaml(fixedyml, TPML_TAGGED_TPM_PROPERTY())
        self.assertEqual(fl[0].property, TPM2_PT.FAMILY_INDICATOR)
        self.assertEqual(fl[0].value, 0x322E3000)
        self.assertEqual(fl[1].property, TPM2_PT.LEVEL)
        self.assertEqual(fl[1].value, 0)
        self.assertEqual(fl[2].property, TPM2_PT.REVISION)
>       self.assertEqual(fl[2].value, 0xA4)
E       AssertionError: 183 != 164

test/test_encoding.py:1160: AssertionError
```

</details>

## Additional context

This was definitely broken somewhere in the #361878 staging train.

## Notify maintainers

@baloo 

---

Note for maintainers: Please tag this issue in your PR.

---

--- a/pkgs/development/python-modules/tpm2-pytss/default.nix
+++ b/pkgs/development/python-modules/tpm2-pytss/default.nix
@@ -4,6 +4,7 @@
   substituteAll,
   buildPythonPackage,
   fetchPypi,
+  fetchpatch,
   pythonOlder,
   asn1crypto,
   cffi,
@@ -37,6 +38,12 @@ buildPythonPackage rec {
     [
       # Fix hardcoded `fapi-config.json` configuration path
       ./fapi-config.patch
+      # libtpms (underneath swtpm) bumped the TPM revision
+      # https://github.com/tpm2-software/tpm2-pytss/pull/593
+      (fetchpatch {
+        url = "https://github.com/tpm2-software/tpm2-pytss/pull/593.patch";
+        hash = "sha256-CNJnSIvUQ0Yvy0o7GdVfFZ7kHJd2hBt5Zv1lqgOeoks=";
+      })
     ]
     ++ lib.optionals isCross [
       # pytss will regenerate files from headers of tpm2-tss.

